services/src/main/java/com/chatbot/services/pubsubservices/PubSubController.java

```diff
@@ -0,0 +1,125 @@
+package com.chatbot.services.pubsubservices;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Map;
+
+import com.chatbot.services.ChatServiceConstants;
+import com.chatbot.services.asyncservices.HangoutsAsyncService;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.ChatClient;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.Event;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.protobuf.Struct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+
+// Handle messages from cloud sub using a push subscriber
+@RestController
+public class PubSubController {
+
+  private HangoutsAsyncService asyncService;
+  private PubSubAuth pubSubAuth;
+  private static final Logger logger = LoggerFactory.getLogger(PubSubController.class);
+
+  public PubSubController(HangoutsAsyncService asyncServiceToSet, PubSubAuth pubSubAuthToSet) {
+    asyncService = asyncServiceToSet;
+    pubSubAuth = pubSubAuthToSet;
+  }
+
+  @PostMapping("/pubsub")
+  String onRequest(@RequestHeader final Map<String, String> headers,
```

Comment:
Why do we need a return value here?

Reply:
empty response body is same as a 200, added comment